inject_metadata.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python3
+
+"""
+Accept a template schema on stdin and inject default metadata fields
+before writing to stdout.
+"""
+
+import json
+import sys
+
+if __name__ == "__main__":
+    # TODO: Actually parse arguments; we currently assume that the filepath
+    # of the schema is passed as argv[1].
+    fileparts = sys.argv[1].split('/')
+
+    # Prepare some default metadata if not overridden in the template schema.
+    bq_dataset = fileparts[0].replace('-', '_')
+    doctype = fileparts[1].replace('-', '_')
+    version = fileparts[2].split('.')[1]
+    bq_table = '{}_v{}'.format(doctype, version)
+
+    # Parse the schema and inject default metadata fields.
+    schema = json.load(sys.stdin)
+    meta = schema.setdefault('mozPipelineMetadata', {})
+    meta.setdefault('bq_dataset', bq_dataset)
```

Comment:
The concept is sound, but I would be careful about naming here, since `bq_dataset` isn't really accurate (there are multiple datasets associated with every `document_namespace`, and each ping is expected to minimally go in `_live` and `payload_bytes_decoded`). I think I'd prefer `document_namespace` or `bq_document_namespace`, which has the added benefit that normalization to bq tables can be done in downstream logic as per SOP.

Reply:
I've updated the proposal to use `bq_dataset_family`.

> normalization to bq tables can be done in downstream logic as per SOP

I rather like the idea of defining the properly normalized bq names at this schema metadata level, which would relive the pipeline of the responsibility of normalizing the names. It certainly makes the connection between doctype and BQ table more explicit.

We do need to be careful not to spread responsibility over too many places for the same thing.